test/lib/sdkdriver/src/testdriver/browser/base.ts

```diff
@@ -0,0 +1,89 @@
+import { EventEmitter } from 'events';
+import { Server as HTTPServer } from 'http';
+import TestDriver from '../';
+import { createWebServer } from '../../util/index';
+
+interface BrowserDriverOptions {
+  browser: 'chrome' | 'firefox';
+  host: string;
+  params?: any;
+}
+
+/**
+ * Browser process driver.
+ * @class
+ * @classdesc A {@link BrowserDriver} manages a browser process where the
+ *   cross-browser test is running. It also manages a web server which serves
+ *   the code that sets up the SDK in the browser.
+ */
+abstract class BrowserDriver extends EventEmitter implements TestDriver {
+  private readonly _webServer: HTTPServer;
+  protected readonly _options: BrowserDriverOptions;
+  abstract startBrowser(url: string): Promise<void>;
+  abstract stopBrowser(): void;
+
+  /**
+   * Constructor.
+   * @param {BrowserDriverOptions} options
+   */
+  constructor(options: BrowserDriverOptions) {
```

Comment:
Probably `sdkUrl` (or whatever we end up calling it) needs to be passed here?

Reply:
I'd prefer it were more strongly typed than `params: any`, WDYT?

Reply:
@markandrus params can be any sdk specific thing, right?

Reply:
Isn't there some set of parameters that aren't SDK-specific, though? For example, we want the BrowserDriver to load a particular module that will exercise the DMP and SDK modules. How do we provide that? Maybe there are already some subsequent commits that address this. Will keep reading